From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: crbednarz <cr.bednarz@gmail.com>
Date: Thu, 2 Jul 2026 18:53:38 -0700
Subject: [PATCH] 9345: Fix nds32 fmul* and fdiv* instructions having incorrect
 fop4

Fix nds32 fmul* and fdiv* instructions having incorrect fop4
 match
---
 Ghidra/Processors/NDS32/data/languages/nds32.sinc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Ghidra/Processors/NDS32/data/languages/nds32.sinc b/Ghidra/Processors/NDS32/data/languages/nds32.sinc
index 6b82fe38eb..a4b4f893f5 100644
--- a/Ghidra/Processors/NDS32/data/languages/nds32.sinc
+++ b/Ghidra/Processors/NDS32/data/languages/nds32.sinc
@@ -1768,12 +1768,12 @@ define pcodeop fnmsubs;
 }
 
 define pcodeop fmuls;
-:fmuls Fst, Fsa, Fsb is $(I32) & $(COP) & fop4=0xa & Fst & Fsa & Fsb  & cop4=0x0 {
+:fmuls Fst, Fsa, Fsb is $(I32) & $(COP) & fop4=0xc & Fst & Fsa & Fsb  & cop4=0x0 {
 	Fst = fmuls(Fsa, Fsb);
 }
 
 define pcodeop fdivs;
-:fdivs Fst, Fsa, Fsb is $(I32) & $(COP) & fop4=0xb & Fst & Fsa & Fsb  & cop4=0x0 {
+:fdivs Fst, Fsa, Fsb is $(I32) & $(COP) & fop4=0xd & Fst & Fsa & Fsb  & cop4=0x0 {
 	Fst = fdivs(Fsa, Fsb);
 }
 
@@ -1890,12 +1890,12 @@ define pcodeop fnmsubd;
 }
 
 define pcodeop fmuld;
-:fmuld Fdt, Fda, Fdb is $(I32) & $(COP) & fop4=0xa & Fdt & Fda & Fdb  & cop4=0x8 {
+:fmuld Fdt, Fda, Fdb is $(I32) & $(COP) & fop4=0xc & Fdt & Fda & Fdb  & cop4=0x8 {
 	Fdt = fmuld(Fda, Fdb);
 }
 
 define pcodeop fdivd;
-:fdivd Fdt, Fda, Fdb is $(I32) & $(COP) & fop4=0xb & Fdt & Fda & Fdb  & cop4=0x8 {
+:fdivd Fdt, Fda, Fdb is $(I32) & $(COP) & fop4=0xd & Fdt & Fda & Fdb  & cop4=0x8 {
 	Fdt = fdivd(Fda, Fdb);
 }
 
-- 
2.45.1

